### ISSUE TITLE
question about cov

### ISSUE THREAD

**Issue by SLAM0102** [Air-IO/Air-IO#21]
Thank you for your reply. I don't have permission to reopen that issue 3 days before. So I create a new issue.
I would like to know:
1. How should the speed uncertainty obtained from network prediction be understood? Can it be considered the same as covariance in nav task?
2. In downstream tasks, should this uncertainty be weighted? Do the weights need to be adjusted manually?

**Comment by haleqiu** [comment]
That's fine.

> Can it be considered the same as covariance in nav task?

What do you mean by the co in nav task?

> In downstream tasks, should this uncertainty be weighted? Do the weights need to be adjusted manually?

For the specific version like EuRoC, we still need to weight it. That is because EuRoC is too small may have overfitting issues.

**Comment by SLAM0102** [comment]
Hey!

The "nav task" I'm referring to concerns the weighting of velocity prediction in integrated navigation systems. For example, when velocity is used as a measurement for updates in EKF, or during graph optimization, how should the weights be determined? Can the uncertainty in the network prediction help?

Additionally, I'd like to know if the sequence length and velocity magnitude affect model performance. As far as I know, for neural networks, more data usually leads to better fitting performance. I wonder if this holds true in time series prediction?

**Comment by haleqiu** [comment]
@SLAM0102 hope this response still help you. The learned cov is helpful, but unfortuneately I thing learnig cov for imu is still challenging, especially when dataset are small. Current the weight is mannually tuned for EuRoC.

For the sequence length, I am not sure. I think it may not be very deterministic.

**Comment by SLAM0102** [comment]
Hey!

I recently conducted experiments on our own collected data (EPSON G366 IMU), including dataset reconstruction and training/inference processes. The AirIO speed prediction results and speed errors for the two sets are as follows:

<img width="3600" height="1800" alt="Image" src="https://github.com/user-attachments/assets/3d40ac06-dbc0-4246-8475-d81d471a4cb1" />

<img width="3600" height="2700" alt="Image" src="https://github.com/user-attachments/assets/102fb82d-58d9-40b2-8ace-f0eb8537d17f" />

<img width="3600" height="1800" alt="Image" src="https://github.com/user-attachments/assets/13323846-610d-4adf-b0e1-8e63b1158c5e" />

<img width="3600" height="2700" alt="Image" src="https://github.com/user-attachments/assets/cdd36a34-72ef-4601-bf08-46512ebcca8e" />

The data length is approximately 20-30 minutes. I just want to know if my experimental results meet the expectations of this network, because the error is much larger compared to EuRoC. Also, do you have any methods to improve accuracy, perhaps maintaining the velocity inference at the sub-meter level for the entire time series?
